op-challenger/game/fault/trace/outputs/provider.go

```diff
@@ -0,0 +1,81 @@
+package outputs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	"github.com/ethereum-optimism/optimism/op-service/client"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	GetStepDataErr      = fmt.Errorf("GetStepData not supported")
+	AbsolutePreStateErr = fmt.Errorf("AbsolutePreState not supported")
+	PreStateRequestErr  = fmt.Errorf("Requested trace index is before prestate block")
+)
+
+type OutputRollupClient interface {
+	OutputAtBlock(ctx context.Context, blockNum uint64) (*eth.OutputResponse, error)
+}
+
+// OutputTraceProvider is a [types.TraceProvider] implementation that uses
+// output roots for given L2 Blocks as a trace.
+type OutputTraceProvider struct {
+	logger        log.Logger
+	rollupClient  OutputRollupClient
+	prestateBlock uint64
+}
+
+func NewTraceProvider(ctx context.Context, logger log.Logger, rollupRpc string, prestateBlock uint64) (*OutputTraceProvider, error) {
+	rollupClient, err := client.DialRollupClientWithTimeout(client.DefaultDialTimeout, logger, rollupRpc)
+	if err != nil {
+		return nil, err
+	}
+	return NewTraceProviderFromInputs(logger, rollupClient, prestateBlock), nil
+}
+
+func NewTraceProviderFromInputs(logger log.Logger, rollupClient OutputRollupClient, prestateBlock uint64) *OutputTraceProvider {
+	return &OutputTraceProvider{
+		logger:        logger,
+		rollupClient:  rollupClient,
+		prestateBlock: prestateBlock,
+	}
+}
+
+func (o *OutputTraceProvider) Get(ctx context.Context, traceIndex uint64) (common.Hash, error) {
+	outputBlock := traceIndex + 1
+	if outputBlock < o.prestateBlock {
+		return common.Hash{}, PreStateRequestErr
+	}
+	output, err := o.rollupClient.OutputAtBlock(ctx, outputBlock)
+	if err != nil {
+		o.logger.Error("Failed to fetch output", "blockNumber", outputBlock, "err", err)
+		return common.Hash{}, err
+	}
+	return common.Hash(output.OutputRoot), nil
+}
+
+// AbsolutePreStateCommitment returns the absolute prestate at the configured prestateBlock.
+// For Optimism Mainnet, this is the first block after bedrock.
```

Comment:
Have we decided on always bisecting to genesis?

Reply:
My understanding is that an output bisection game of depth 63 would _by far_ sufficiently cover blocks from genesis. @clabby had told me it should cover blocks until heat death I believe haha.

But anyways, I'm pretty strictly following what Adrian had outlined in https://github.com/ethereum-optimism/client-pod/issues/41. That way I wouldn't surprise him when he gets back :)

Reply:
Yeah an output bisection game over a depth of 63 should cover us until heat death, but it's way overkill to have a tree that large for the outputs. The tentative plan is to split the tree of d=63, There is also a question of where exactly we split it.

If we're only bisecting between the previous finalized output root and the current one, we don't need much depth on that portion of the bisection at all, but if we are going all the way from genesis, we'll need a depth with enough padding to support us for a long time going forwards. For the instruction trace bisection, assuming we're only executing a single block, a depth of 40 supporting $2^{40}$ instructions should be sufficient w/ some extra padding, which would imply the tree needs to be deeper than 63 if we're bisecting from genesis, since the top layer of bisection could only support 8,388,608 blocks. 

Will look something like this:
```
                   o
                 /    \
                o      o
               / \    / \
 ---------------- [SPLIT] ----------------
             o     o o    o
.......................................... and so on
```

Reply:
Assuming we go from the previous finalized output, the output bisection tree depth also places an implicit max on the proposal submission interval; Should be concious of that while designing and consider OP Stack chains with sparse proposals. Doesn't seem like a reasonable bet to assume that the proposal submission interval is small everywhere.

Reply:
Yeah I thought we went with bisecting from genesis so that we aren't forced to post output roots at any regular interval. @karlfloersch had pretty strong feelings about it.  It may be ok if we can start from an arbitrarily old previous finalized output but I'm not sure that buys us anything since we'd have to allow for it to be really stupid old anyway and so still have about the same depth as we would from genesis.

Benefit of not starting from genesis though is that you then don't need an archive node.  But we need an archive node to create the cannon traces so it's not necessarily a win.

Reply:
I think the code here works the same regardless of where we start the bisection anyway given it's just given a prestate block number, that could either be the bedrock activation block or the block the last finalized output root is for and it just works.

I'd probably just remove the reference to OP Mainnet here and leave the trace provider to be completely agnostic to how the start block is chosen.